lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1105,10 +1113,17 @@ void TNonblockingServer::listenSocket(THRIFT_SOCKET s) {
   serverSocket_ = s;
 
   if (!port_) {
-    sockaddr_in addr;
-    socklen_t size = sizeof(addr);
-    if (!getsockname(serverSocket_, reinterpret_cast<sockaddr*>(&addr), &size)) {
-      listenPort_ = ntohs(addr.sin_port);
+    #ifdef _WIN32
+      struct sockaddr_storage addr;
+      struct sockaddr_in *pAddr = (sockaddr_in *) &addr;
+      socklen_t size = sizeof(addr);
+    #else
+      struct sockaddr_in addr;
+      struct sockaddr_in *pAddr = &addr;
+      socklen_t size = sizeof(addr);
+    #endif
```

Comment:
Sure, and on Unix, the result shall be truncated rather than fail if the size is not enough.
What I was wondering is whether the failure is official Windows API spec or MinGW only behavior.